uamqp/message.py

```diff
@@ -458,25 +458,32 @@ def _multi_message_generator(self):
 
         :rtype: generator[~uamqp.message.Message]
         """
+        unappended_message_bytes = None
         while True:
             new_message = self._create_batch_message()
             message_size = new_message.get_message_encoded_size() + self.size_offset
             body_size = 0
+            if unappended_message_bytes:
+                new_message._body.append(unappended_message_bytes)
+                body_size += len(unappended_message_bytes)
             try:
                 for data in self._body_gen:
-                    message_segment = []
-                    if isinstance(data, six.text_type):
-                        data = data.encode(self._encoding)
-                    batch_data = c_uamqp.create_data(data)
-                    c_uamqp.enocde_batch_value(batch_data, message_segment)
-                    combined = b"".join(message_segment)
-                    body_size += len(combined)
+                    message_bytes = None
+                    if not isinstance(data, Message):  # raw data
```

Comment:
After giving it some thought - I think this type check could be easier to adapt to the EH SDK if we do it like this:
```python
try:
    if not data.application_properties:  # Message-like object
        data.application_properties = self.application_properties
    message_bytes = data.encode_message()
except AttributeError:  # raw data
    wrap_message = Message(body=data, application_properties=self.application_properties)
    message_bytes = wrap_message.encode_message()
```
This means we could consume any "message-like" object, so long as it has an "application_properties" attribute and an "encode_message" function. This should make it easier to adapt the EH SDK, as an `Event` object already has the `application_properties` attribute, we just need to give it an `encode_message` function and we should be able to use it directly.

Reply:
Great idea! Get what you mean and it reminds me of duck typing.
This is undoubtedly a better implementation. I think what we should do after taking this implementation is to provide more detailed documentations about batching in EventData.